keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -192,584 +173,1331 @@ logs are lost.
-network is down for the rest of the containers in the pod.
-
-This has another major disadvantage: several apps crash if traffic is blackholed
-during startup (common in some rails middleware, for example) and have to resort
-to some kind of workaround, like [this one][linkerd-wait] to wait. This makes
-also service mesh miss their goal of augmenting containers functionality without
-modifying the main application.
-
-This KEP addresses these 3 problems just listed when initContainer are not used
-by the application. If initContainers are used, the first and the last problem
-are solved only. In other words, traffic might still be blackholed for
-initContainers that run after the service mesh iptables rules are inserted.
-
-Such rules are usually inserted as an initContainer (trying to run last, to
-avoid blackholing traffic to other initContainers) or alternatively, in the case
-of Istio, using a [CNI plugin][istio-cni-opt]. When using the CNI plugin all
-traffic from initContainers will be blackholed.
-
-[istio-cni-opt]: https://istio.io/latest/docs/setup/additional-setup/cni/
-[linkerd-wait]: https://github.com/olix0r/linkerd-await
-
-##### Istio bug report
-
-There is also a [2 years old bug][istio-bug-report] from Istio devs that this
-KEP will fix. In addition, similar benefit is expected for Linkerd, as we talked
-with Linkerd devs.
-
-One of the things mentioned there is that, at least in 2018, a workaround used
-was to tell the user to run a script to wait for the service mesh to start on
-their containers.
-
-Rodrigo will reach out to Istio devs to see if the situation changed since 2018.
-
-[istio-bug-report]: https://github.com/kubernetes/kubernetes/issues/65502
-
-#### Move containers out of the pod
-
-Due to the ordering problems of having the container in the same pod, another
-option is to move it out of the pod. This will, for example, remove the problem
-of shutdown order. Furthermore, Rodrigo will argue than in many cases this is
-better or more elegant in Kubernetes.
-
-While some things might be possible to move to daemonset or others, it is not
-possible for all applications. For example some apps are not multi-tenant and
-this can not be an option security-wise. Or some apps would still like to have a
-sidecar that adds some metadata, for example, or just uses a simple sidecar
-(like file uploading after containers finish in a Job, or create/update some
-files in shared emptyDir volumes, etc.).
-
-While this is an option, is not possible or extremely costly for several use
-cases.
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Pods that had sidecars will need to be deleted and recreated without them.
+
+The feature gate disablement will require the kubelet restart. When kubelet will
+start, it will fail to reconcile the Pod with new fields and will clean up all
+running containers.
```

Comment:
Assuming it's a release in which it understand the new field, right?
Otherwise it will just kill sidecars and leave everything else, right?

Reply:
this section is about disabling feature. Downgrade will require node drain so this will not be an issue

Reply:
OK. Can you clarify it in the text? Sth like adding:

If version downgrade is involved, drain is required as described in Downgrade strategy section above?

Reply:
Not applied - please don't resolve without applying the comment.